### PR TITLE
Add floating labels effect to "inside" label position

### PR DESCRIPTION
**Pull request by truongwp**
Inline documentation is in progress.

Fixes https://github.com/Strategy11/formidable-pro/issues/3617

### CODE REVIEW

File: js/formidable.js [hunk 1]
```diff
@@ -1135,6 +1138,129 @@ function frmFrontFormJS() {
...
+			if ( isIE() ) {
+				document.querySelectorAll( selector ).forEach( function( input ) {
+					checkPlaceholderIE( input );
+				});
+			}
+		};
+
+		runOnLoad();
+
+		jQuery( document ).on( 'frmPageChanged', function( event ) {
```
Comment (Crabcyborg) [inline js/formidable.js:1259]: Does `documentOn` work here too by any chance?

Comment (truongwp) [reply js/formidable.js:1259]: I tried this before but `document.addEventListener()` and `documentOn()` don't work with `jQuery( document ).trigger()`;

**Comment by Crabcyborg** [comment]
@garretlaxton this update is mostly done and could use some solid testing. It updates the "Inside" label position setting so that the label will transition from the input to the label position with an animation.

File: js/formidable.js [hunk 2]
```diff
@@ -1153,7 +1303,7 @@ function frmFrontFormJS() {
 			jQuery( document.getElementById( 'frm_resend_email' ) ).on( 'click', resendEmail );
 
 			jQuery( document ).on( 'change', '.frm-show-form input[name^="item_meta"], .frm-show-form select[name^="item_meta"], .frm-show-form textarea[name^="item_meta"]', frmFrontForm.fieldValueChanged );
-			jQuery( document ).on( 'change keyup', '.frm-show-form .frm_inside_container input, .frm-show-form .frm_inside_container select, .frm-show-form .frm_inside_container textarea', maybeShowLabel );
+			// jQuery( document ).on( 'change keyup', '.frm-show-form .frm_inside_container input, .frm-show-form .frm_inside_container select, .frm-show-form .frm_inside_container textarea', maybeShowLabel );
```
Comment (Crabcyborg) [inline js/formidable.js:1306]: We probably just want to remove this line if we don't need it anymore. Keeping commented out code usually just confuses things.

File: tests/fields/test_FrmFieldsController.php [hunk 3]
```diff
@@ -18,7 +18,10 @@ public function test_prepare_placeholder() {
 			'required'    => 0,
 		);
 		$placeholder = $this->prepare_placeholder( $field );
-		$this->assertEquals( $name, $placeholder, 'an empty string should be replaced by the label inside of an input.' );
+		// $this->assertEquals( $name, $placeholder, 'an empty string should be replaced by the label inside of an input.' );
```
Comment (Crabcyborg) [inline tests/fields/test_FrmFieldsController.php:21]: This line too, I think we can just remove the commented assert.

**Comment by truongwp** [comment]
@garretlaxton Thanks for testing. I fixed those issues. Please check again.

**Review by garretlaxton — APPROVED** [review]
This looks great now!

**Comment by Crabcyborg** [comment]
@truongwp I'm still seeing the overlapping issue when I refresh like I mention at the end of https://github.com/Strategy11/formidable-forms/pull/791#issuecomment-1160656751

I played with it for a moment. I have an incomplete update here that tries to correct any inside containers that might have a value on load. https://github.com/Strategy11/formidable-forms/pull/803

I checked Chrome and it looks like it doesn't have this behavior. But this is how it works in Firefox if you enter values into a field, then refresh the page.

**Pull request merged**